pkgs/dash_analytics/test/dash_analytics_test.dart

```diff
@@ -871,4 +853,75 @@ $initialToolName=${ConfigHandler.dateStamp},$toolsMessageVersion
     expect(query2.flutterChannelCount, 1,
         reason: 'The first instance has flutter information initialized');
   });
+
+  test('Payload sent to GA follows limitations', () {
```

Comment:
nit prefer many small unit tests with descriptive test names rather than long tests with many expectations with descriptive reasons. For example, I'd break this up into the following tests:

- "a request does not allow more than 25"
- "a request allows 25 events"
- "an event does not allow 26 parameters"
- "an event allows 25 parameters"
etc

Reply:
For context, having long tests with lots of expectations means that previous code could have unexpected side effects on later, unrelated expectations

Reply:
(I'm sure this code is correct today, I'm more looking forward to 2 years from now, when there's 10x more code)

Reply:
You're definitely right, I'm going to break these tests up within a test group and create an issue to breakup this large file into logical test files as well for a future PR